pkg/controllers/provisioning/scheduling/suite_test.go

```diff
@@ -3778,6 +3778,478 @@ var _ = Context("Scheduling", func() {
+			Expect(node.Labels).ToNot(HaveKey(cloudprovider.ReservationIDLabel))
+			Expect(node.Labels).To(HaveKeyWithValue(v1.CapacityTypeLabelKey, Not(Equal(v1.CapacityTypeReserved))))
+			Expect(node.Labels).To(HaveKeyWithValue(corev1.LabelInstanceTypeStable, "small-instance-type"))
+		})
+		It("should correctly track reservations for multiple nodepools with distinct reservations for the same instance pool", func() {
+			nodePool.Name = "np-1"
+			nodePool2 := test.NodePool(*nodePool, v1.NodePool{ObjectMeta: metav1.ObjectMeta{Name: "np-2"}})
+			ExpectApplied(ctx, env.Client, nodePool, nodePool2)
+
+			// Give the second NodePool the same instance types as the first, except overwrite the reservation ID for the target
+			// instance type. This test should verify that the scheduler treats these offerings as though they are drawing from
+			// two separate pools.
+			//cloudProvider.InstanceTypesForNodePool[nodePool.Name] = append([]*cloudprovider.InstanceType{}, cloudProvider.InstanceTypes...)
+			distinctInstanceType := fake.NewInstanceType(fake.InstanceTypeOptions{
+				Name: "small-instance-type",
+				Resources: map[corev1.ResourceName]resource.Quantity{
+					corev1.ResourceCPU:    resource.MustParse("2"),
+					corev1.ResourceMemory: resource.MustParse("2Gi"),
+				},
+			})
+			distinctInstanceType.Offerings = append(distinctInstanceType.Offerings, &cloudprovider.Offering{
+				ReservationCapacity: 1,
+				Available:           true,
+				Requirements: pscheduling.NewLabelRequirements(map[string]string{
+					v1.CapacityTypeLabelKey:     v1.CapacityTypeReserved,
+					corev1.LabelTopologyZone:    "test-zone-1",
+					v1alpha1.LabelReservationID: "r-distinct",
+				}),
+				Price: fake.PriceFromResources(distinctInstanceType.Capacity) / 100_000.0,
+			})
+			distinctInstanceType.Requirements.Get(v1.CapacityTypeLabelKey).Insert(v1.CapacityTypeReserved)
+			cloudProvider.InstanceTypesForNodePool[nodePool2.Name] = []*cloudprovider.InstanceType{distinctInstanceType}
+
+			pods := lo.Times(2, func(i int) *corev1.Pod {
+				return test.UnschedulablePod(test.PodOptions{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{
+							"app": "test",
+						},
+					},
+					PodAntiRequirements: []corev1.PodAffinityTerm{{
+						LabelSelector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								"app": "test",
+							},
+						},
+						TopologyKey: corev1.LabelHostname,
+					}},
+					NodeRequirements: []corev1.NodeSelectorRequirement{
+						{
+							Key:      corev1.LabelInstanceTypeStable,
+							Operator: corev1.NodeSelectorOpIn,
+							Values:   []string{distinctInstanceType.Name},
+						},
+						{
+							Key:      v1.NodePoolLabelKey,
+							Operator: corev1.NodeSelectorOpIn,
+							Values:   []string{fmt.Sprintf("np-%d", i+1)},
+						},
+					},
+				})
+			})
+
+			// Even though the pods schedule to separate NodePools, those NodePools share a capacity reservation for the
```

Comment:
I think this comment is not correct given the code below

Reply:
Good catch, you're right. Will update